lib/ansible/modules/cloud/hcloud/hcloud_location_facts.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_location_facts
+
+short_description: Gather facts about your Hetzner Cloud locations.
+
+version_added: "2.8"
+
+description:
+    - Gather facts about your Hetzner Cloud locations.
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the location you want to get.
+        type: int
+    name:
+        description:
+            - The name of the location you want to get.
+        type: str
+extends_documentation_fragment: hcloud
+"""
+
+EXAMPLES = """
+- name: Gather hcloud location facts
+  local_action:
+    module: hcloud_location_facts
+
+- name: Print the gathered facts
+  debug:
+    var: ansible_facts.hcloud_location_facts
+"""
+
+RETURN = """
+ansible_facts:
```

Comment:
The return docs should be changed as follows:

~~~
hcloud_location_facts:
    description: The location facts as list
    returned: always
    type: complex
    contains:
        id: 
            description: Numeric identifier of the location
            returned: always
            type: int
            sample: 1937415
        name:
            description: Name of the location
            returned: always
            type: str
            sample: fsn1
        description:
            description: Detail description of the location
            returned: always
            type: str
            sample: Falkenstein DC Park 1
        country:
            description: Country code of the location
            returned: always
            type: str
            sample: DE
        city:
            description: City of the location
            returned: always
            type: str
            sample: Falkenstein
~~~